packit/specfile.py

```diff
@@ -240,3 +255,43 @@ def get_source(self, source_name: str) -> Optional[Tag]:
         # sanitize the name, this will also add index if there isn't one
         source_name, *_ = Tags._sanitize_tag(source_name, 0, 0)
         return next(self.tags.filter(name=source_name, valid=None), None)
+
+    def read_patch_comments(self):
+        """Read the spec again, detect comment lines right above a patch-line
+        and save it as an attribute to the patch for later retrieval.
+
+        Match patch-lines with the patch-data from rebase-helper on the name of
+        the patches.
+        """
+        comment = []
+        patch_comments = {}
+        for line in self.spec_content.section("%package"):
+            if not line.strip():
+                comment = []
+            if line.startswith("#"):
+                comment.append(line.removeprefix("#").strip())
+            if line.lower().startswith("patch"):
+                patch_name = Path(line.split(":", 1)[1].strip()).name
+                patch_comments[patch_name] = comment
+                comment = []
+        for kind in ["applied", "not_applied"]:
+            self.patches[kind] = [
+                Patch(patch, patch_comments.get(patch.get_patch_name(), []))
+                for patch in self.patches[kind]
+            ]
+
+    def remove_patches(self):
+        """Remove all patch-lines from the spec file"""
+        content = []
+        stretch = []
+        for line in self.spec_content.section("%package"):
+            stretch.append(line)
+            if not line.strip():
+                content += stretch
+                stretch = []
+            if line.lower().startswith("patch"):
+                stretch = []
+                if not content[-1].strip():
+                    content.pop()
```

Comment:
I may eventually find how the method works, but some more inline comments would be welcome, thanks!